labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,7 +60,15 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case s: String => s"A string with length ${s.length()}"
+      case i: Int if i > 0 => "A positive integer"
+      case p: Person => s"A person with name: ${p.name}"
+      case seq: Seq[_] if seq.length > 10 => "Seq with more than 10 elements"
+      case seq: Seq[_] if seq.length >= 3 => s"first: ${seq(0)}, second: ${seq(1)}, rest: ${seq.drop(2)}"
+      case null => "A null value"
+      case _ => "Some Scala class"
+    }
```

Comment:
Good call naming the case variables different things. I dislike that you used 's' 'i' and 'p' :+1:  :-1: